### PR TITLE
Suggestions for GSdx CRC changes.

### PR DESCRIPTION
**Pull request by pgert**

### CODE REVIEW

**Comment by gregory38** [comment]
@ramapcsx2 could you give a look at those change?

**Comment by bositman** [comment]
This looks good, one question are the regions of the games all confirmed? If so I think after the conflict is resolved it's fine to merge

**Comment by avih** [comment]
What's the reason for this change? just more correct regions? or should/does it also affect the hacks which are applied to different games/regions? (some crc hacks have region-specific code).

**Comment by bositman** [comment]
It's a cleanup, it disabled some duplicate codes, changed all no region entries to the correct region and added some comments about which exact game the hack is for (see metal gear series). No hacks have actually been changed/removed (except for duplicates which won't affect anything)

**Comment by avih** [comment]
OK, thanks. So let him rebase and let's merge.

Also, please provide more meaningful commit message such as `gsdx: crc hacks: regions cleanups (should not affect functionality)`

File: plugins/GSdx/GSCrc.cpp [hunk 1]
```diff
@@ -236,7 +241,7 @@ CRC::Game CRC::m_games[] =
 	{0x56242EC9, Genji, KO, 0},
 	{0xCDAF243D, Genji, CH, 0}, 
 	{0x2A5E0B61, Genji, CH, 0},
-	{0x7D4EA48F, Genji, NoRegion, 0},
+	// {0x7D4EA48F, Genji, EU, 0}, // same CRC as {0x7D4EA48F, HauntingGround, EU, 0},
```
Comment (avih) [inline plugins/GSdx/GSCrc.cpp]: Don't change this. `0x7D4EA48F` was a duplicate for hunting ground and genji, commit 0dc1e7f4bba63f28bbb5cbf32bc7ea647f2ee97d commented the hunting ground one - leaving it only for genji, so why did you now comment this line - which removes `0x7D4EA48F` completely so it doesn't get used at all?

File: plugins/GSdx/GSCrc.cpp [hunk 2]
```diff
@@ -116,20 +118,22 @@ CRC::Game CRC::m_games[] =
...
-	{0x947B933B, OnePieceGrandAdventure, US, 0},
+	{0xE1674F57, OnePieceGrandBattle, EU, 0},
 	{0xB049DD5E, OnePieceGrandBattle, US, 0},
+	{0xCA2073B3, OnePieceGrandBattle, KO, 0},
 	{0x5D02CC5B, OnePieceGrandBattle, NoRegion, 0},
 	{0x6F8545DB, ICO, US, 0},
 	{0xB01A4C95, ICO, JP, 0},
 	{0x2DF2C1EA, ICO, KO, 0},
-	{0x5C991F4E, ICO, NoRegion, 0},
-	{0x7ACF7E03, ICO, NoRegion, 0},
```
Comment (avih) [inline plugins/GSdx/GSCrc.cpp]: Same as the previous comment for `0x7ACF7E03` - leave it for ICO.

**Comment by pgert** [comment]
These are the duplicates affected:

-	{0x7ACF7E03, ICO, NoRegion, 0},
+	// {0x7ACF7E03, ICO, NoRegion, 0}, // same CRC as "Twisted Metal - Black" (PAL)
+	//  and as {0x7ACF7E03, SpyroNewBeginning, NoRegion, 0},

-	{0x1A85E924, GodOfWar, NoRegion, 0}, // cutie comment
+	// {0x1A85E924, GodOfWar, NoRegion, 0}, // cutie comment
+	//  same CRC as {0x1A85E924, DevilMayCry3, CH, 0},

-	{0x7D4EA48F, Genji, NoRegion, 0},
+	// {0x7D4EA48F, Genji, EU, 0}, // same CRC as {0x7D4EA48F, HauntingGround, EU, 0},

If two (or more) discs of different games have the same CRC, is it better
 to have the (assumably) most common one enabled instead of disabling all of them?
It is incorrect, but OK - What should it be?

**Comment by avih** [comment]
It should be what it was, There were no duplicates before your patch. Those which were duplicates in the past - one of the duplicates was commented out and had a comment that it's a duplicate. The other (presumably the more popular one) stayed uncommented and effective.

Also, you've added several CRCs. Could you please list:
1. Which CRCs were added (where did you get those from BTW?)
2. Which CRCs were removed, if at all, and why.
3. Any other modifications.

And also provide a meaningful commit message which explain what you did and why.
Thanks.

**Comment by pgert** [comment]
Code modified:
ICO, God of War & Genji has been re-enabled.

"Which CRCs were added (where did you get those from BTW?)"
It's pretty easy to spot the added CRCs.
The CRCs was found in the WS-archive (devina40), and on various sites through Google.
The sources seemed reliable.
I won't make a list of each added CRC and the source from where it was fetched.
No CRCs were removed.
Other modifications: Region-codes (won't make a list).

File: plugins/GSdx/GSState.cpp [hunk 3]
```diff
@@ -3413,7 +3413,7 @@ bool GSC_Spartan(const GSFrameInfo& fi, int& skip)
 {
 	if(skip == 0)
 	{
-		if(g_crc_region == CRC::NoRegion &&fi.TME && fi.FBP == 0x02000 && fi.FPSM == PSM_PSMCT32 && fi.TBP0 == 0x00000 && fi.TPSM == PSM_PSMCT32)
+		if(g_crc_region == CRC::EU &&fi.TME && fi.FBP == 0x02000 && fi.FPSM == PSM_PSMCT32 && fi.TBP0 == 0x00000 && fi.TPSM == PSM_PSMCT32)
```
Comment (avih) [inline plugins/GSdx/GSState.cpp]: What's the purpose of this change?

**Comment by pgert** [comment]
"{0x72E1E60E, Spartan, NoRegion, 0}," has been changed to "{0x72E1E60E, Spartan, EU, 0},",
 and "{0x08277A9E, Spartan, US, 0}," has been added.
The hack is specifically for 72E1E60E (or assumably to the PAL region),
 and therefore "NoRegion" has been changed to "EU".

**Comment by ssakash** [comment]
then, what about the US version ? does this crc hack only work for EU version ? in that case, It is correct.

File: plugins/GSdx/GSState.cpp (hunk 3 above)
Comment (ssakash) [inline plugins/GSdx/GSState.cpp]: the former region value of that game at GSCrc.cpp was NoRegion after updating the region to EU. he also had to change this to EU to make the CRC hack accessible for the PAL version. if this isn't changed to EU then, the CRC hack will not be accessible for the PAL version.

**Comment by avih** [comment]
@pgert, please do the following and I'll merge it:
1. Split it into two commits: 1. non functional changes in one commit (move stuff around, updated comments, etc). 2. functional changes.
2. Please add meaningful commit messages (mostly for the one with the functional changes).
3. if required, squash such that the PR ends up with these two commits only.

**Comment by pgert** [comment]
@ssakash:
As I undestand it, the hack is not intended for the US or JP regions.
I guess that porting the hack to these regions can be done (but not by me).

**Comment by ssakash** [comment]
Okay, that's fine. Follow avih's instructions to get the PR merged.

**Comment by ssakash** [comment]
@pgert
if you don't know how to squash the commits, avih could do it for you. just give a word to know.

**Comment by pgert** [comment]
Code modified.
This should be enough.

New CRCs:
	{0xBE6A9CFB, DBZBT2, NoRegion, 0},
	{0xCE93CB30, DBZBT3, JP, 0},
	{0xE1674F57, OnePieceGrandBattle, EU, 0},
	{0x3B0ADBEF, Manhunt2, US, 0},
	{0xF2A8D307, GodOfWar, US, 0},
	{0x9536E111, NamcoXCapcom, JP, 0},
	{0xBF0DC4CE, TenchuFS, DE, 0},
	{0xC8DADF58, TenchuWoH, EU, 0},
	{0x6B149273, SilentHill2, EU, 0},
	{0x6BBD4932, SilentHill2, EU, 0}, // Director's Cut

Various corrections of disc-region.

No changes in comments except for 891F223F & 086273D2 to explain why the discs is of the EU region, and not (exclusively) of the FR region.
No entries moved around.

**Comment by ssakash** [comment]
nice. if you have finalized the changes then, you can squash it now.

**Comment by pgert** [comment]
Squash?

**Comment by ssakash** [comment]
yes, use the Github rebase command and pick out the 5 latest commits and squash them together into a single commit.

https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by pgert** [comment]
@avih:
It doesn't seem that squashing can be made without installing GitHub on ones own computer - that it can't be done "in the cloud".
I can't successfully install GitHub, so can you make use of this anyway - otherwise I can scrap this PR and make a new one that looks like this one does now.

**Comment by avih** [comment]
@pgert 
> It doesn't seem that squashing can be made without installing GitHub on ones own computer - that it can't be done "in the cloud".

I believe that's true. I also think that as a developer it's probably best to be able to run git locally* since interacting with the repo online has limitations, and the local git access gives you much more control and tools.

So I'd suggest to give it yet another go. You can use a "portable" version of git by downloading `PortableGit-1.9.5-preview20150319.7z` from here https://github.com/msysgit/msysgit/releases , extracting it and running the batch file. It will NOT mess with your system and should work out of the box (that's how I use git on Windows).

If you prefer to still not do it locally now, you can either create a clean new PR with a single commit (and useful commit comment), or I can help you and do it for you - this time. Next time I'd expect you to be able to do that on your own ;)

Just let me know which one you choose.

**Comment by avih** [comment]
@pgert so, few things now that the patch is easier to follow:

1. The patch does not apply cleanly to latest PCSX2 source code. It (the commutative patches) does apply cleanly to a 5ed3f5f1a4e7fd6a2180a855b4ca59004fb6c160 from early March, but applying it to that revision, then quashing all the commits to a single patch and then trying to apply it to the latest revision still fails. So I'm not really sure how GitHub says above "This pull request can be automatically merged.", because I think it can't. If anyone could help me understand this, please say so.

So I'd need you to rebase it such that it applies cleanly to the latest PCSX2.

2. What's the reason for changing the regions? is it for cosmetic/correctness reasons? Or did you intend to change some behavior (how?) ?

3. I see that you modified the spartan crc hack to correctly use the new region. I'm assuming you did check for every other region which you changed that the hack does not depend on region, but I'd prefer to hear you say so explicitly instead of me only assuming this.

**Comment by ssakash** [comment]
1) If you want, I could try rebasing it and sending PR to make it apply cleanly to the latest PCSX2.
2) the older regions were Invalid/duplicate, the changes were a sort of a clean up for correctness.
3) your assumption is correct. every region has a different CRC so, it couldn't be applied to any other regions. this CRC hack was only focused for the PAL version.

**Comment by avih** [comment]
@ssakash, unless you're the same guy as @pgert , I was addressing him.

**Comment by pgert** [comment]
@avih:
For simplicity, I'm closing this PR and replacing it with a new one.
